### PR TITLE
Adding updates to packages to update SqlClient in the 1.1.x release

### PR DESCRIPTION
**Pull request by saurabh500**
The changes enable the build of the service release for System.Data.SqlClient for release/1.1.0 branch of .Net Core. 

Verified locally by `build.cmd` at the root of the clone. 
Following is a structure of what was produced

```
F:\NETCORE\COREFX\BIN\PACKAGES\DEBUG
|   Microsoft.Private.PackageBaseline.1.0.1-servicing-25127-0.nupkg
|   System.Data.SqlClient.4.3.1-servicing-25127-0.nupkg
|   System.Diagnostics.DiagnosticSource.4.3.1-servicing-25127-0.nupkg
|   System.Net.Http.4.3.2-servicing-25127-0.nupkg
|   System.ValueTuple.4.3.1-servicing-25127-0.nupkg
|
+---reports
|       Microsoft.Private.PackageBaseline.json
|       System.Data.SqlClient.json
|       System.Diagnostics.DiagnosticSource.json
|       System.Net.Http.json
|       System.ValueTuple.json
|
+---specs
|       Microsoft.Private.PackageBaseline.nuspec
|       Microsoft.Private.PackageBaseline.nuspec.pkgpath
|       System.Data.SqlClient.nuspec
|       System.Data.SqlClient.nuspec.pkgpath
|       System.Diagnostics.DiagnosticSource.nuspec
|       System.Diagnostics.DiagnosticSource.nuspec.pkgpath
|       System.Net.Http.nuspec
|       System.Net.Http.nuspec.pkgpath
|       System.ValueTuple.nuspec
|       System.ValueTuple.nuspec.pkgpath
|
\---symbols
        System.Data.SqlClient.4.3.1-servicing-25127-0.symbols.nupkg
        System.Diagnostics.DiagnosticSource.4.3.1-servicing-25127-0.symbols.nupkg
        System.Net.Http.4.3.2-servicing-25127-0.symbols.nupkg
        System.ValueTuple.4.3.1-servicing-25127-0.symbols.nupkg
```

There is a System.Data.SqlClient.4.3.1-servicing-25127-0.nupkg package with the updated package version. 
Verified that the runtime DLL has the required changes that were meant to go into service release.

### CODE REVIEW

**Comment by ericstj** [comment]
You're getting packaging errors:
```
18:17:13 D:\j\workspace\windows_nt_de---62a15db0\Tools\Packaging.targets(1077,5): error : PackageIndex from D:\j\workspace\windows_nt_de---62a15db0\pkg\Microsoft.Private.PackageBaseline\packageIndex.json indicates that assembly version 4.1.1.1 is contained in non-stable package version 4.3.1 which differs from this package version 4.3.0. [D:\j\workspace\windows_nt_de---62a15db0\src\System.Data.SqlClient\pkg\System.Data.SqlClient.pkgproj]
18:17:13 D:\j\workspace\windows_nt_de---62a15db0\Tools\Packaging.targets(1077,5): error : PackageIndex from D:\j\workspace\windows_nt_de---62a15db0\pkg\Microsoft.Private.PackageBaseline\packageIndex.json is has an assembly version entry(s) for 4.1.1.0 which are no longer in package System.Data.SqlClient.  Please run /t:UpdatePackageIndex on this project to commit an update. [D:\j\workspace\windows_nt_de---62a15db0\src\System.Data.SqlClient\pkg\System.Data.SqlClient.pkgproj]
```

File: src/System.Data.SqlClient/dir.props [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.1.1.0</AssemblyVersion>
+    <AssemblyVersion>4.1.1.1</AssemblyVersion>
```
Comment (ericstj) [inline src/System.Data.SqlClient/dir.props]: You're missing the update to package version.

**Comment by gkhanna79** [comment]
@ericstj @weshaggard Are you good with this change?

File: src/System.Data.SqlClient/dir.props (hunk 1 above)
Comment (saurabh500) [inline src/System.Data.SqlClient/dir.props]: I just pushed in a commit to fix the package errors. 
Added `<PackageVersion>4.3.1</PackageVersion>` to dirs.props

**Comment by ericstj** [comment]
```
12:39:22 /mnt/resource/j/workspace/dotnet_corefx/release_1.1.0/centos7.1_debug_prtest/Tools/Packaging.targets(1077,5): error : System.Data.SqlClient should support API version 4.1.1.0 on .NETStandard,Version=v1.3 but System.Data.SqlClient/ref/netstandard1.3/System.Data.SqlClient.dll was found to support 4.1.0.0. [/mnt/resource/j/workspace/dotnet_corefx/release_1.1.0/centos7.1_debug_prtest/src/System.Data.SqlClient/pkg/System.Data.SqlClient.pkgproj]
12:39:22 /mnt/resource/j/workspace/dotnet_corefx/release_1.1.0/centos7.1_debug_prtest/Tools/Packaging.targets(1077,5): error : System.Data.SqlClient should support API version 4.1.1.0 on .NETStandard,Version=v1.5 but System.Data.SqlClient/ref/netstandard1.3/System.Data.SqlClient.dll was found to support 4.1.0.0. [/mnt/resource/j/workspace/dotnet_corefx/release_1.1.0/centos7.1_debug_prtest/src/System.Data.SqlClient/pkg/System.Data.SqlClient.pkgproj]
```

The error is telling you that your PKGProj implies you want an assembly with version 4.1.1.0 to be applicable when targeting NS1.3 or NS1.5, but your package will only have an assembly with version 4.1.0.0.

This is happening because your desktop ref was versioned but the NS one was not (automatically by packaging).

In master we’ve suppressed this with the following:
https://github.com/dotnet/corefx/blob/327e743692a7f3ed09baaa0bc91cbdd5b6ac73aa/Packaging.props#L32-L37

You can add that to your pkgproj to fix the problem.

**Comment by saurabh500** [comment]
@ericstj The page hadn't refreshed and I saw your reply after posting the question. Thanks. I will try out the recommendation.

**Comment by saurabh500** [comment]
@ericstj Do the changes look OK?

**Comment by ericstj** [comment]
Changes look ok, be sure to test your built package from an application.

**Comment by paalmoest** [comment]
When can I except these changes to be available in any official release ? And what are other ways to get this into our runtime ?

**Pull request merged**